helper/validation/write_only.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-cty/cty"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// PreferWriteOnlyAttribute is a ValidateResourceConfigFunc that returns a warning
+// if the Terraform client supports write-only attributes and the old attribute is
+// not null.
+// The last step in the path must be a cty.GetAttrStep{}.
+// When creating a cty.IndexStep{} to into a nested attribute, use an unknown value
+// of the index type to indicate any key value.
+// For lists: cty.Index(cty.UnknownVal(cty.Number)),
+// For maps: cty.Index(cty.UnknownVal(cty.String)),
+// For sets: cty.Index(cty.UnknownVal(cty.Object(nil))),
+func PreferWriteOnlyAttribute(oldAttribute cty.Path, writeOnlyAttributeName string) schema.ValidateResourceConfigFunc {
```

Comment:
Should `writeOnlyAttributeName` be a cty.Path as well?

Reply:
I originally had this as a `cty.Path` but I changed it since the only thing we would be doing with the path is getting the attribute name. I didn't want the parameter type to give the impression that we are actually walking/validating this path, but we can change it back to a `cty.Path` for future proofing.